plugins/UserId/Archiver.php

```diff
@@ -0,0 +1,130 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\UserId;
+use Piwik\Container\StaticContainer;
+use Piwik\DataArray;
+use Piwik\DataTable;
+use Piwik\Metrics;
+
+/**
+ * Archiver that aggregates metrics per user ID (user_id field).
+ */
+class Archiver extends \Piwik\Plugin\Archiver
+{
+    const USERID_ARCHIVE_RECORD = "UserId_archive_record";
```

Comment:
@mattab do you maybe have a good suggestion for the archive record name of the UserId report? Maybe `UserId_users`?